design/authenticators_health_api.md

```diff
@@ -0,0 +1,233 @@
+it if it's not needed).
+
+This feature lets the person who configures the Authenticator to get an immediate feedback 
+on the configuration, before any user needs to run an authentication request.
+
+For example, the OIDC Authenticator configuration includes whitelisting the authenticator in the CONJUR_AUTHENTICATORS variable:
+
+`CONJUR_AUTHENTICATORS=authn-oidc/adfs,authn`
+
+Let's say the Conjur Operator configured the OIDC Authenticator and forgot to whitelist 
+the authenticator in the CONJUR_AUTHENTICATORS variable. Without the Helthcheck API, 
+the first time the user will encounter the error will be in an authentication request. 
+A user `alice` will try to authenticate with OIDC and will get a 401 Unauthorized 
+response without any description on what happened (the answer will be in the logs which 
+are probably unaccessible to her) even if the actual request is ok. The main problem 
+here is that the Unauthorized response has nothing to do with the actual request, as 
+the problem is not in it. The problem is in the configuration which was made by another 
+person in another time. With the Healthcheck API the person who does the configuration will 
+get an immediate feedback on it, and the users that are *using* the authenticators will not 
+be unauthorized due to configuration errors.
+
+### Current Mechanism
+
+The current mechanism for understanding that there is a problem in the Authenticator configuration is as follows:
+
+1. The Operator configures the Authenticator according to the docs
+1. A user tries to authenticate\
+    1. this may be the Operator himself imitating a user to verify the authenticator works in the dev env
+1. The user gets a 401 Unauthorized response
+    1. There are other response codes possible but configuration issues will be 401
+1. The user reports to the operator that there is an issue
+    1. This one is tricky as the user might not know who to report this to.
+    1. Again, this may be the operator himself
+1. The operator looks in the logs (according to the docs) and finds the issue
+1. The operator fixes the issue
+
+This can be iterative as there might be several issues. It is possible to assume that in the case 
+above, the operator will solve this with the user and will ask him to try to authenticate again and 
+if it's still Unauthorized will solve the configuration issues one-by-one. 
+
+### Process Logic
+
+1. The Conjur operator logs into Conjur
+1. The Conjur operator configures the Authenticators according to the docs
+    1. We might have several authenticators configured
+1. The operator runs the Healthcheck request: `GET /authenticators/health`
+1. The operator gets a response:
+    1. In case all the authenticators are healthy, a response with code 200 and the following body:
+        
+        ```
+        {
+           “authenticators”:
+           {
+              "authn-1": "ok",
+              "authn-2": "ok"
+           },
+           "ok": true
```

Comment:
should this be `"status"` instead of `"ok"` as above?

Reply:
i was following the behaviour of our general health-check to not break ux:
```
{
  "services": {
    "ldap-sync": "ok",
    "possum": "ok",
    "ui": "ok",
    "ok": true
  },
  "database": {
    "ok": true,
    "connect": {
      "main": "ok"
    },
    "free_space": {
      "main": {
        "kbytes": 63672,
        "inodes": 128876
      }
    },
    "replication_status": {
      "pg_current_xlog_location": "0/23ECDD0",
      "pg_current_xlog_location_bytes": 37670352
    }
  },
  "ok": true
}
```

Reply:
oh i see what happened now. i took srdjan's suggestion without looking at the status. i'll change it in the doc to be like the health-check